src/components/trade/PriceSuggestions/PriceSuggestionItem.tsx

```diff
@@ -28,28 +29,39 @@ interface FormattedPrices {
 
 const LOW_PRICE_FLOOR = new BigNumber('0.0001')
 
-interface FormatPriceOptions {
-  useThreshold?: boolean | undefined
-  decimals?: number | undefined
+function formatPriceWithFloor(price: BigNumber): string {
+  const priceAsBN = parseAmount(price.toString(10), 18)
+  if (!priceAsBN || priceAsBN.isZero()) return 'N/A'
+
+  const displayPrice = formatAmount({ amount: priceAsBN, precision: 18, decimals: DEFAULT_DECIMAL_PLACES })
+  return price.gt(LOW_PRICE_FLOOR) ? displayPrice : '< ' + LOW_PRICE_FLOOR.toString(10)
 }
 
-export function formatPriceWithFloor(
-  price: BigNumber,
-  { useThreshold = false, decimals = PRICE_ESTIMATION_PRECISION }: FormatPriceOptions,
-): string {
-  const priceAsBN = parseAmount(price.toString(10), 18)
-  return price.gt(LOW_PRICE_FLOOR) || !useThreshold
-    ? // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
-      formatAmount({ amount: priceAsBN!, precision: 18, decimals })
-    : '< ' + LOW_PRICE_FLOOR.toString()
+interface GetPriceFormatted {
+  price: BigNumber | null
+  isPriceInverted: boolean
 }
 
-function getPriceFormatted(price: BigNumber | null, isPriceInverted: boolean): FormattedPrices {
+function getPriceFormatted({ price, isPriceInverted }: GetPriceFormatted): FormattedPrices {
   if (price) {
-    const inversePriceLabel = formatPriceWithFloor(invertPrice(price), { useThreshold: true })
-    const priceLabel = formatPriceWithFloor(price, { useThreshold: true })
-    const inversePriceValue = invertPrice(price).toString(10)
-    const priceValue = price.toString(10)
+    const inversePriceLabel = formatPriceWithFloor(invertPrice(price))
+    const priceLabel = formatPriceWithFloor(price)
+    // Use full precision for accuracy and tightest price
+    const inversePriceBN = parseAmount(invertPrice(price).toString(10), 18) as BN
+    const priceValueBN = parseAmount(price.toString(10), 18) as BN
+
+    // Have to explicitly pass full precision when using object params
+    const inversePriceValue = formatAmountFull({
+      amount: inversePriceBN,
+      precision: 18,
+      thousandSeparator: false,
+    })
```

Comment:
Actually I think we don't have to, but that's how we set up the types :man_shrugging: